streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala

```diff
@@ -143,6 +149,21 @@ class ReceiverTracker(ssc: StreamingContext, skipReceiverLaunch: Boolean = false
    */
   private val receiverPreferredLocations = new HashMap[Int, Option[String]]
 
+  /**
+   * The max timeout to launch a receiver. If a receiver cannot register in time, StreamingContext
+   * will be stopped.
+   */
+  private val RECEIVER_LAUNCHING_MAX_TIMEOUT =
```

Comment:
Timeout itself means a max time limit the system will wait. What is the meaning of a max timeout?